pkg/ds/daemon_set.go

```diff
@@ -410,24 +465,53 @@ func (ds *daemonSet) WatchDesires(
 				if ds.Disabled {
 					continue
 				}
-				err = ds.handleNodeChanges(labeledChanges)
+
+				err = ds.removePods()
 				if err != nil {
+					err = util.Errorf("Unable to remove pods from intent tree: %v", err)
+					continue
+				}
+				addedNodes, err = ds.addPods()
+				if err != nil {
+					err = util.Errorf("Unable to add pods to intent tree: %v", err)
 					continue
 				}
+				nodesToAdd <- addedNodes
 
 			case <-timer.C:
+				if ds.Disabled {
+					paused = true
+					continue
+				}
+
+				if paused {
+					unpauseReplication <- struct{}{}
```

Comment:
doesn't this have the side effect of unpausing replication? It seems at adds with the comment below

Reply:
it's the comment that's wrong, it does make sense to unpause here because the daemon set is not disabled (since we would have hit the `continue` above otherwise)

Reply:
I think technically i shouldn't have to check pausing/unpausing here because that would only happen on the update channel, but there already was a disabled check so just keeping in that spirit to change as few things as possible